include/oneapi/mkl/detail/backend_selector.hpp

```diff
@@ -0,0 +1,71 @@
+/*******************************************************************************
+* Copyright 2020 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions
+* and limitations under the License.
+*
+*
+* SPDX-License-Identifier: Apache-2.0
+*******************************************************************************/
+
+#ifndef _ONEMKL_BACKEND_SELECTOR_HPP_
+#define _ONEMKL_BACKEND_SELECTOR_HPP_
+
+#include "oneapi/mkl/exceptions.hpp"
+#include "oneapi/mkl/detail/backends.hpp"
+#include "oneapi/mkl/detail/get_device_id.hpp"
+
+namespace oneapi {
+namespace mkl {
+
+template <backend Backend>
+class backend_selector {
+public:
+    explicit backend_selector(sycl::queue queue) : queue_(queue) {
+        if ((queue.is_host() || queue.get_device().is_cpu())) {
+            if (Backend != backend::mklcpu) {
+                throw unsupported_device("", "backend_selector", queue.get_device());
```

Comment:
Do you think we can provide more information backend_selector, e.g. "backend_selector created for Backed"?

Reply:
I am proposing to do it in this format then: e.g. "backend_selector<backend::mklcpu>"

Reply:
Agree.